### ISSUE TITLE
Upgrade to V2 API

### ISSUE THREAD

**Issue by jordan8037310** [singer-io/tap-harvest#9]
V1 API from Harvest is now deprecated: https://www.getharvest.com/blog/2017/09/introducing-harvest-api-v2/

It will be supported until January 2019. Will use this ticket to track effort to upgrade.

**Comment by nickfogler** [comment]
Any thoughts about upgrading to v2?  I'd really like to see the Invoice Line Items included (https://help.getharvest.com/api-v2/invoices-api/invoices/invoices/#the-invoice-line-item-object).

**Comment by suzybates** [comment]
I would also like to see the "fixed fee" fields on the Project and the Project Role data working.

**Comment by awarzon** [comment]
Thanks for this! I'm testing the latest release since it was pushed last week and ran into an error, see below. It appears that Harvest allows strings into the `external_reference.group_id` field and the Github/Harvest integration puts string data in.

```
2018-08-26 21:20:55,308Z    tap - CRITICAL Errors during transform
2018-08-26 21:20:55,308Z    tap - 	group_id: awme does not match {'type': ['integer', 'null']}
2018-08-26 21:20:55,308Z    tap - 	: {'id': '35', 'group_id': 'awme', 'service': 'github.com', 'permalink': 'https://github.com/<REMOVED>/issues/35', 'service_icon_url': 'https://proxy.harvestfiles.com/production_harvestapp_public/uploads/platform_icons/github.com.png?1349823336'} does not match {'properties': {'id': {'type': ['string', 'null']}, 'permalink': {'type': ['string', 'null']}, 'service_icon_url': {'type': ['string', 'null']}, 'task_id': {'type': ['null', 'integer']}, 'group_id': {'type': ['integer', 'null']}, 'service': {'type': ['string', 'null']}}, 'type': 'object'}

```

**Comment by dmosorast** [comment]
@awarzon Thanks for posting! These kinds of unexpected data/schema issues are being shaken out right now, so this is very helpful to keep the testing moving forward.

I've changed the schema of `group_id` to be a string in light of this ([#24](https://github.com/singer-io/tap-harvest/pull/24)), and will release it in version 2.0.4 shortly.

**Comment by tsmith512** [comment]
Related to #16 , time entry sync doesn't propagate deletions. I'm in the process of trying to write a little side cron job that will inspect the API response and delete rows out of our Stitch-populated tables that have been deleted in Harvest, but my approach seems brittle. I haven't read up much on the APIv2 yet, but is there a way deletions could be handled in this upgrade?

**Comment by dmosorast** [comment]
Closing this as V2 has been merged in #20
